### DISCUSSION TITLE
Best way to load CSV from HTTP API

### DISCUSSION THREAD

**Discussion by helpermethod** [Homebrew/discussions#2816] [Tap maintenance and Homebrew development] [answered]
Hi,

sorry for this probably stupid question (I'm a Homebrew noob), but what is the best way to obtain a CSV file during the installation process?

I've seen that you can define `resource`s, but it seems they serve a different purpose.

So I guess shelling out to curl would be the better solution?

**Comment by SMillerDev** [comment]
A resource can be whatever you want it to be, including a csv. We just rarely need a csv.

See resources with `homebrew-` as the name prefix for some random downloads (usually for tests)

**Comment by carlocab** [comment, marked as answer]
You definitely want a `resource` here. `resource`s are cached and checksummed, not to mention are backed by code that make downloads more resilient against connection errors, etc.

You could probably shell out to `curl` and then do all that after, but why duplicate all the work that's been done for you?

**Reply by helpermethod** [reply]
Hi @carlocab!

That sounds great! Do I need to explicitly set the checksum? The contents of the CSV file may change multiple times per day.

**Reply by SMillerDev** [reply]
Yes, Homebrew resources need to be checksummed otherwise you can never ensure that you install it the way you packaged it.

**Reply by helpermethod** [reply]
I guess in that case I need to use something else because the response of the API changes too often.

**Reply by SMillerDev** [reply]
If this is for inclusion in Homebrew/core or cask that might not be acceptable. If it's for your own tap I'd recommend just using curl like you mentioned.
